spec/lib/topic_query_spec.rb

```diff
@@ -919,6 +919,22 @@ class ::TopicQuery
       end
     end
 
+    context 'when preloading associations' do
+      it "preloads associations" do
+        TopicList.topic_preloader_associations << :first_post
+
+        topic = Fabricate(:topic)
+        Fabricate(:post, topic: topic)
+
+        new_topic = topic_query.list_new.topics.first
+        expect(new_topic.association(:image_upload).loaded?).to eq(true) # Preloaded by default
+        expect(new_topic.association(:first_post).loaded?).to eq(true)   # Testing a user-defined preloaded association
+        expect(new_topic.association(:user).loaded?).to eq(false)        # Testing the negative
+
+        TopicList.topic_preloader_associations.clear
```

Comment:
I think this would clear the default preloader associations, which could cause issues for future tests?